docs/tutorials/TracingCode.adoc

```diff
@@ -0,0 +1,270 @@
+the execution path _inside_ each component. By hiding those details, the digram succeeds in informing the reader
+about the
+overall execution path of a command without overwhelming the reader with too much details. In this tutorial, you aim to
+find those omitted details so that you get a more in-depth understanding of the code base.
+
+Before we proceed, ensure that you have done the following:
+
+. Read the _Architecture_ section of the DG
+. Set up the project in Intellij IDEA
+. Learn basic debugging features of Intellij IDEA
+
+== Setting a break point
+
+As you know, the first step of debugging is to put in a breakpoint where you want the debugger to pause the execution.
+For example, if you are trying to understand how the App starts up, you would put a breakpoint in the first statement
+of the `main` method. In our case, we would want to begin the tracing at the very point where the App start processing
+user input i.e., somewhere in the UI component, and then trace through how the execution proceeds through the UI component.
+However, that the execution path through a GUI is often somewhat obscure due to various _even-driven mechanisms_
+used by GUI frameworks, which happens to be the case here too. Therefore, let us put the breakpoint where the UI transfers
+control to the Logic component. According to the sequence diagram, the UI component yields control to the Logic component
+through a method named `execute`. Searching through the code base for `execute()` yields a promising candidate in
+`seedu.address.ui.CommandBox.CommandExecutor`.
+
+.Using the `Search for target by name` feature. `Navigate` > `Symbol`.
+image::Execute.png[width=1011px, height=372px]
+{empty} +
+
+A quick look at the class confirms that this is indeed close to what we're looking for. However, it is just an
+`Interface`. Let's delve further and find the implementation of the interface by using the `Find Usages` feature in
+IntelliJ IDEA.
+
+.`Find Usages` tool window. `Edit` > `Find` > `Find Usages`.
+image::FindUsages.png[width=968px, height=394px]
+{empty} +
+
+Bingo! `MainWindow#executeCommand()` seems to be exactly what we're looking for!
+
+Now let's set the breakpoint. First, double-click the item to reach the corresponding code. Once there, click on the left
+gutter to set a breakpoint, as shown below.
+image:LeftGutter.png[width=1171px, height=154px]
+
+
+[qanda]
+
+
+== Tracing the execution path
+
+Recall from the User Guide that the `edit` command has the format: `edit INDEX [n/NAME] [p/PHONE] [e/EMAIL]
+[a/ADDRESS] [t/TAG]...` For this tutorial we will be issuing the command `edit 1 n/Alice Yeoh`.
+
+[TIP]
+.Taking notes
+====
+Over the course of the debugging session, you will encounter every major component in the application. Try to jot down
+what happens inside the component and where the execution transfers to another component.
+====
+
+. To start the debugging session, simply `Run` > `Debug `Main`
```

Comment:
There is an extra back-tick in the rendered view.

Reply:
Removed.